mod-pubsub-server/src/main/java/org/folio/config/user/SystemUserConfig.java

```diff
@@ -1,29 +1,58 @@
 package org.folio.config.user;
 
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Component;
 
 import io.vertx.core.json.JsonObject;
+import lombok.Getter;
 
 @Component
 public class SystemUserConfig {
+  private static final Logger LOGGER = LogManager.getLogger();
+  private static final String SYSTEM_USER_NAME_VAR = "SYSTEM_USER_NAME";
+  private static final String SYSTEM_USER_PASSWORD_VAR = "SYSTEM_USER_PASSWORD";
+
+  @Getter
   private final String name;
   private final String password;
 
-  public SystemUserConfig(@Value("${SYSTEM_USER_NAME:pub-sub}") String name,
-    @Value("${SYSTEM_USER_PASSWORD:pubsub}") String password) {
+  public SystemUserConfig(@Value("${SYSTEM_USER_NAME:#{null}}") String name,
+    @Value("${SYSTEM_USER_PASSWORD:#{null}}") String password) {
 
+    validateCredentials(name, password);
     this.name = name;
     this.password = password;
   }
 
-  public String getName() {
-    return name;
-  }
-
   public JsonObject getUserCredentialsJson() {
     return new JsonObject()
       .put("username", name)
       .put("password", password);
   }
+
+  private static void validateCredentials(String name, String password) {
+    LOGGER.info("validateCredentials:: validating system user credentials");
+    List<String> missingVariables = new ArrayList<>();
+    if (name == null) {
```

Comment:
I might also check that the strings are not blank (empty string)!

Reply:
done, thanks